File: frontend/src/types/audit.ts
export interface AuditLog {
  id: string;
  event_id: string;
  actor_user_id: string | null;
  action: string;
  entity_type: string;
  entity_id: string | null;
  case_id: string | null;
  timestamp: string | null;
  ip_address: string | null;
  metadata_json: Record<string, any> | null;
  previous_event_hash: string;
  event_hash: string;
}

export interface AuditVerifyOut {
  valid: boolean;
  events_checked: number;
  broken_at_index: number | null;
}

export interface SignatureOut {
  id: string;
  document_id: string;
  version_id: string;
  signer_id: string;
  document_hash: string;
  algorithm: string;
  is_mock: boolean;
  signed_at: string | null;
}

export interface SignatureVerifyOut {
  valid: boolean;
  is_mock: boolean;
  disclaimer: string;
}

File: app/schemas/audit.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class AuditLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    event_id: str
    actor_user_id: UUID | None
    action: str
    entity_type: str
    entity_id: str | None
    case_id: UUID | None
    timestamp: datetime | None
    ip_address: str | None
    metadata_json: dict | None
    previous_event_hash: str
    event_hash: str


class AuditVerifyOut(BaseModel):
    valid: bool
    events_checked: int
    broken_at_index: int | None


class SignatureOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    document_id: UUID
    version_id: UUID
    signer_id: UUID
    document_hash: str
    algorithm: str
    is_mock: bool
    signed_at: datetime | None


class SignatureVerifyOut(BaseModel):
    valid: bool
    is_mock: bool
    disclaimer: str
